compiler/frontend/src/hir.rs

```diff
@@ -272,6 +332,7 @@ impl Debug for IdKey {
                 disambiguator,
             } => {
                 write!(f, "{name}")?;
+                // TODO(Optimization): Only include the `#` in Debug output and not when formatting for the HirId path
```

Comment:
Not sure if I understand this correctly. Do you want to drop all disambiguators in the `IdPath`? Because that would break things (we could no longer differentiate multiple assignments with the same name in the same scope)

Reply:
You are right, that Idea does not hold considering my last assumption was wrong.